edsdme/blocks/partners-navigation/features/search/gnav-search.js

```diff
@@ -129,10 +130,34 @@ class Search {
 
   getSuggestions(query = this.query) {
     const { env } = getConfig();
-    const subdomain = env === 'prod' ? 'adobesearch' : 'adobesearch-stage';
-    const api = `https://${subdomain}.adobe.io/autocomplete/completions?q[locale]=${locale.ietf}&scope=${CONFIG.suggestions.scope}&q[text]=${encodeURIComponent(query)}`;
-
-    return fetch(api, { headers: { 'x-api-key': CONFIG.suggestions.apiKey } })
+    const partnerDataCookie = getPartnerDataCookieObject(getCurrentProgramType());
+    const partnerLevel = partnerDataCookie?.level?.toLowerCase() || 'public';
+    const regions = partnerDataCookie?.permissionRegion?.toLowerCase() || 'worldwide';
+    const specializations = partnerDataCookie?.permissionSpecializations?.toLowerCase();
+    const { locales } = getConfig();
+    const localesData = getLocale(locales);
+    let domain = 'https://io-partners-dx.stage.adobe.com';
+    if (env.name === 'prod') {
+      domain = 'https://io-partners-dx.adobe.com';
+    }
+    const url = new URL(
+      `${domain}/api/v1/web/dx-partners-runtime/search-apc/search-apc?`,
+    );
+    const queryParams = new URLSearchParams();
+    queryParams.append('suggestions', true);
+    queryParams.append('partnerLevel', partnerLevel);
+    queryParams.append('regions', regions);
+    queryParams.append('specializations', specializations);
+    queryParams.append('term', query);
+    queryParams.append('geo', localesData.prefix && localesData.region);
+    queryParams.append('language', localesData.ietf);
+    queryParams.append('size', SUGGESTIONS_SIZE);
+    url.search = queryParams.toString();
+    const headers = new Headers();
+    headers.append('Content-Type', 'application/json');
+    headers.append('Authorization', 'Basic NDA3M2UwZTgtMTNlMC00ZjZjLWI5ZTMtZjBhZmQwYWM0ZDMzOjJKMnY1ODdnR3dtVXhoQjNRNlI2NDIydlJNUDYwRDZBYnJtSzRpRTJrMDBmdlI1VGMxRXNRbG9Vc2dBYTNNSUg=');
+
+    return fetch(url.toString(), { headers })
       .then((data) => data.json())
```

Comment:
it would be nice to have some utils for generation of all fetch query params since we already have this on one more place